seccomp/src/lib.rs

```diff
@@ -1154,58 +1154,44 @@ mod tests {
     const KVM_GET_PIT2_MSB: u64 = 0x0000_ae9f;
     const KVM_GET_PIT2_LSB: u64 = 0x8070_0000;
 
-    const EXTRA_SYSCALLS: [i64; 5] = [
-        libc::SYS_rt_sigprocmask,
-        libc::SYS_sigaltstack,
-        libc::SYS_munmap,
-        libc::SYS_exit,
-        libc::SYS_rt_sigreturn,
-    ];
-
     fn validate_seccomp_filter(
         rules: Vec<(i64, Vec<SeccompRule>)>,
         validation_fn: fn(),
         should_trigger_sigsys: bool,
     ) {
-        let mut filter =
-            SeccompFilter::new(rules.into_iter().collect(), SeccompAction::Kill).unwrap();
+        let filter = SeccompFilter::new(rules.into_iter().collect(), SeccompAction::Kill).unwrap();
 
-        let triggered_sigsys: Arc<AtomicBool> = Arc::new(AtomicBool::new(true));
+        let triggered_sigsys: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
         let shared_triggered_sigsys = triggered_sigsys.clone();
 
-        // We need 2 threads here: in case of a seccomp denial, the inner thread will be killed
-        // and the outter thread will fail. The execution will be returned to the instruction
-        // that follows the outter thread `join()` method.
-        let outter_thread = thread::spawn(move || {
-            let inner_thread = thread::spawn(move || {
-                // whitelist needed syscalls
-                for syscall in EXTRA_SYSCALLS.iter() {
-                    assert!(filter
-                        .add_rules(
-                            *syscall,
-                            vec![SeccompRule::new(vec![], SeccompAction::Allow)],
-                        )
-                        .is_ok());
-                }
+        // We need to execute the seccomp thread inside a catch_unwind block in order to
+        // avoid printing unneeded warnings in case of a seccomp denial
+        let seccomp_result = panic::catch_unwind(|| {
+            // We need to execute the seccomp validation fn inside a different thread.
+            // In case of a seccomp denial this thread will fail.
+            thread::spawn(move || {
                 // apply filter
                 assert!(filter.apply().is_ok());
 
+                // set triggered_sigsys to true before calling the validation fn
```

Comment:
nit: Upper case the first letter and use full stop in comments.

Reply:
Done.